ambry-api/src/main/java/com/github/ambry/account/AccountService.java

```diff
@@ -107,6 +107,37 @@
    */
   public boolean removeAccountUpdateConsumer(Consumer<Collection<Account>> accountUpdateConsumer);
 
+  /**
+   * Add a collection of containers to an existing account.
+   * @param accountName the name of account in which new container will be added.
+   * @param containers a collection of containers to add.
+   * @return a collection of containers that are successfully added. If original input containers includes existing one,
+   *         the existing containers will be ignored and not present in the result.
+   */
+  default Collection<Container> addContainers(String accountName, Collection<Container> containers) {
```

Comment:
I think these methods should throw exceptions (might need to create one).  They can't return false, and having the impl throw IllegalStateException on any failure doesn't feel right.

Reply:
Yes, I was trying to use RestException but felt not appropriate. I can introduce new type exception, for example, `AccountServiceException`. What do you think?